nvdaHelper/remote/vbufRemote.cpp

```diff
@@ -16,30 +16,39 @@ This license can be found at:
 #include "vbufRemote.h"
 #include <vbufBase/backend.h>
 #include "dllmain.h"
+#include <common/log.h>
 
 using namespace std;
 
-map<VBufBackend_t*,HINSTANCE> backendLibHandles;
+const map<wstring,VBufBackend_create_proc> VBufBackendFactoryMap {
+	{L"adobeAcrobat",AdobeAcrobatVBufBackend_t_createInstance},
+	{L"adobeFlash",AdobeFlashVBufBackend_t_createInstance},
+	{L"gecko_ia2",GeckoVBufBackend_t_createInstance},
+	{L"mshtml",MshtmlVBufBackend_t_createInstance},
+	{L"lotusNotesRichText",lotusNotesRichTextVBufBackend_t_createInstance},
+	{L"webKit",WebKitVBufBackend_t_createInstance}
+};
 
 extern "C" {
 
 VBufRemote_bufferHandle_t VBufRemote_createBuffer(handle_t bindingHandle, int docHandle, int ID, const wchar_t* backendName) {
-	wchar_t backendPath[MAX_PATH];
-	wsprintf(backendPath,L"%s\\VBufBackend_%s.dll",dllDirectory,backendName);
-	HINSTANCE backendLibHandle=LoadLibrary(backendPath);
-	if(backendLibHandle==NULL) return NULL;
-	VBufBackend_create_proc createBackend=(VBufBackend_create_proc)GetProcAddress((HMODULE)(backendLibHandle),"VBufBackend_create");
-	if(createBackend==NULL) {
-		FreeLibrary(backendLibHandle);
-		return NULL;
-	} 
+
+	auto i=VBufBackendFactoryMap.find(backendName);

```

Comment:
If `backendName` is `nullptr` then I don't think it will convert to a `wstring`

```suggestion
	if(!backendName){
		LOG_ERROR(L"BackendName is nullptr");
		return nullptr;
	}
	auto i=VBufBackendFactoryMap.find(backendName);
```